env0/utils.go

```diff
@@ -75,7 +75,7 @@ func readResourceData(i interface{}, d *schema.ResourceData) error {
 				field.Set(reflect.ValueOf(sshKeys))
 			}
 		case reflect.String, reflect.Bool, reflect.Int:
-			field.Set(reflect.ValueOf(dval))
+			field.Set(reflect.ValueOf(dval).Convert(fieldType))
```

Comment:
This supports "casting" from one type to the other when both types are of the same kind.
E.g.:
```
type MyString string
```